lib/src/edhoc.rs

```diff
@@ -483,60 +492,86 @@ pub fn i_process_message_2(
             let plaintext_2_decoded = decode_plaintext_2(&plaintext_2, plaintext_2_len);
 
             if plaintext_2_decoded.is_ok() {
-                let (c_r_2, kid, mac_2, ead_2) = plaintext_2_decoded.unwrap();
+                let (c_r_2, id_cred, mac_2, ead_2) = plaintext_2_decoded.unwrap();
                 c_r = c_r_2;
 
                 // Step 3: If EAD is present make it available to the application
-                let ead_success = if let Some(ead_2) = ead_2 {
-                    i_process_ead_2(ead_2, cred_r_expected, &h_message_1).is_ok()
+                let (ead_ok, r_authenticated_via_ead, cred_r) = if let Some(ead_2) = ead_2 {
+                    // if EAD-zeroconf is present, then id_cred must contain a full credential
```

Comment:
Ideally, this should check if the EAD handler is the zero touch draft handler. No action needed at this point, let's track.

Reply:
Tracked in #120.